core/src/main/scala/io/finch/Input.scala

```diff
@@ -103,17 +103,17 @@ object Input {
       Input(copied, i.route)
     }
 
-    def apply[F[_]: Effect, S[_[_], _], A](s: S[F, A])(implicit
+    def apply[F[_]: Dispatcher, S[_[_], _], A](s: S[F, A])(implicit
```

Comment:
Same question here:

I think we should keep asking for Async type-class and derive Dispatcher out of it later.